facilitator/src/http.rs

```diff
@@ -25,23 +26,112 @@ impl Method {
     }
 }
 
-/// Creates an agent with a timeout of 10 seconds
-pub(crate) fn create_agent() -> Agent {
-    AgentBuilder::new().timeout(Duration::from_secs(10)).build()
+/// An HTTP agent that can be configured to manage "Authorization" headers and
+/// retries using exponential backoff.
+#[derive(Debug, Clone)]
+pub(crate) struct RetryingAgent {
+    /// Agent to use for constructing HTTP requests.
+    agent: Agent,
+    /// By default, requests which fail due to transport problems or which
+    /// return any 5xx HTTP code will be retried with exponential backoff. If
+    /// set, these HTTP codes will also cause retries.
+    additional_retryable_http_codes: Option<Vec<u16>>,
```

Comment:
`Vec<u16>` seems enough, since the vec can be empty.

Reply:
Excellent suggestion, especially given how it simplifies `is_http_status_retryable`.